src/lib/optimizer/abstract_syntax_tree/abstract_ast_node.hpp

```diff
@@ -218,6 +191,57 @@ class AbstractASTNode : public std::enable_shared_from_this<AbstractASTNode> {
 
   // @{
   /**
+   * Altering the order of Columns in a subtree.
+   *
+   * The order of Columns in a subtree might change, e.g., as the result of Join Re-Ordering. In this call all nodes
+   * above the root of the JoinTree need to adapt the ColumnIDs they refer to. Think, e.g. the ColumnIDs of a SortNode.
+   *
+   * In order to this:
+   *    1. Obtain the ColumnOrigins of the *old* subtree-root by calling get_column_origins() on it.
```

Comment:
What is the old, what is the new root? I think this would be much easier to understand with an example - either ASCII art or in the wiki. If you're interested, we could also talk about this on the blog.